### PR TITLE
ux update

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
First update to the UX of the tool

1. Running the binary (“depstat”) directly prints help (available options).
2. “help” sub command prints help.
3. “list” subcommand lists all dependencies
4. “cycles” subcommand prints all the cycles in the dependencies
5. “stats”  subcommand prints the metrics: total number of dependencies, max depth of dependencies, number of transitive dependencies.
        a. “-v” verbose flag : lists all dependencies and the longest dependency chain
        b. “--file” : produces a JSON file having the metrics at the specified location, if no location specified then the file is created in the current directory

image

### CODE REVIEW

File: cmd/cycles.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			cycles = append(cycles, actualCycle)
+		}
+	}
+	return cycles
+}
+
+func init() {
+	rootCmd.AddCommand(cyclesCmd)
+
+	// Cobra supports Persistent Flags which will work for this command
```
Comment (alenkacz) [inline cmd/cycles.go]: you should drop this probably

File: cmd/list.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	for _, dep := range deps {
+		fmt.Println(dep)
+	}
+	fmt.Println()
+}
+
+func init() {
+	rootCmd.AddCommand(listCmd)
+
+	// Cobra supports Persistent Flags which will work for this command
```
Comment (alenkacz) [inline cmd/list.go]: same here

File: cmd/stats.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	// for A -> B -> C the depth is 2
+	return maxDeps - 1
+}
+
+func init() {
+	rootCmd.AddCommand(statsCmd)
+	statsCmd.Flags().BoolP("verbose", "v", false, "Get additional details")
+	statsCmd.Flags().StringVarP(&fileLocation, "file", "f", "", "Direct the output to a file")
+
+	// Cobra supports Persistent Flags which will work for this command
```
Comment (alenkacz) [inline cmd/stats.go]: and here :)

File: cmd/stats.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		maxDeps = max(maxDeps, deps)
+	}
+	// for A -> B -> C the depth is 2
+	return maxDeps - 1
+}
+
+func init() {
+	rootCmd.AddCommand(statsCmd)
+	statsCmd.Flags().BoolP("verbose", "v", false, "Get additional details")
+	statsCmd.Flags().StringVarP(&fileLocation, "file", "f", "", "Direct the output to a file")
```
Comment (alenkacz) [inline cmd/stats.go]: nit: this could be the --output flag we talked about :)

**Review by alenkacz — APPROVED** [review]
Looks great, I have only nits. Thanks 👏

**Comment by RinkiyaKeDad** [comment]
Thanks for the review @alenkacz! Made the changes, please take a look :)

**Review by alenkacz — APPROVED** [review]
🚢 🎉

**Pull request merged**